BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift

```diff
@@ -79,24 +73,22 @@ public final class ConvivaAnalytics: NSObject {
         self.customerKey = customerKey
         self.config = config
 
-        let systemInterFactory: CISSystemInterfaceProtocol = IOSSystemInterfaceFactory.initializeWithSystemInterface()
-        let setting: CISSystemSettings = CISSystemSettings()
-
-        logger = Logger(loggingEnabled: config.debugLoggingEnabled)
-        self.contentMetadataBuilder = ContentMetadataBuilder(logger: logger)
-
         if config.debugLoggingEnabled {
-            setting.logLevel = LogLevel.LOGLEVEL_DEBUG
-        }
-
-        let systemFactory = CISSystemFactoryCreator.create(withCISSystemInterface: systemInterFactory, setting: setting)
-        let clientSetting: CISClientSettingProtocol = try CISClientSettingCreator.create(withCustomerKey: customerKey)
-        if let gatewayUrl = config.gatewayUrl {
-            clientSetting.setGatewayUrl(gatewayUrl.absoluteString)
+            var settings = [String: Any]()
+            if let gatewayUrl = config.gatewayUrl {
+                settings[CIS_SSDK_SETTINGS_GATEWAY_URL] = gatewayUrl.absoluteString
```

Comment:
Don't think it's a problem, personally. We only set the gateway url in debug situations. The crux of the matter is that I simply carried over the functionality from before - ie. that integration debug logging is linked to Conviva debug logging. So the question is whether you want to disconnect those two debug logging types, given that it will differ from the previous version if we do that.

Reply:
Are we ok with leaving this as is or should I change it?

Reply:
@sinewave440hz , sorry I could not get back on this. 

> We only set the gateway url in debug situations

Thanks for sharing, wasn't aware of this.

> The crux of the matter is that I simply carried over the functionality from before - ie. that integration debug logging is linked to Conviva debug logging. So the question is whether you want to disconnect those two debug logging types, given that it will differ from the previous version if we do that.

I did not get this part. In the original code, the 2 settings are independent. The 2 settings(ConvivaSDK log level and gatewayURL) are set independent of each other and gateway URL setting is also independent of `config.debugLoggingEnabled` setting. Can you please clarify?

        if config.debugLoggingEnabled {
            setting.logLevel = LogLevel.LOGLEVEL_DEBUG
        }
        
        if let gatewayUrl = config.gatewayUrl {
            clientSetting.setGatewayUrl(gatewayUrl.absoluteString)
        }

Reply:
Sorry I think I didn't explain this clearly enough. You are right that gateway url is set independently of logging situations. What I meant is that logging in the integration is linked to Conviva logging in the previous version: `logger = Logger(loggingEnabled: config.debugLoggingEnabled)` It is still possible to use the gateway url without debugging simply by passing in a different `LogLevel`. So we don't lose any functionality, it's just packaged differently.

Reply:
We could absolutely split them - ie. make gateway url settable independent of the integration's logging setting. While it's not that likely AFAIK that a single customer will want to set several different urls, it may be useful to be able to see integration debug even when using the (automatically assigned) production gateway url. But then again, that implies testing on a production Conviva environment. This is why I went with the current approach. Do you understand what I mean? And if so, what do you think?

Reply:
Will do!

Reply:
Added that now. I also made it possible to set the Conviva log level, which was previously fixed to warning level.